okio/jvm/jvm.gradle

```diff
@@ -4,10 +4,20 @@ apply plugin: 'ru.vyarus.animalsniffer'
 kotlin.targets.matching { it.platformType.name == 'jvm' }.all { target ->
   target.project.sourceCompatibility = JavaVersion.VERSION_1_7
   target.project.targetCompatibility = JavaVersion.VERSION_1_7
-
-  tasks['jar'].configure {
-    manifest {
-      attributes('Automatic-Module-Name': 'okio')
+  
+  // configure the kotlin JVM jar task
+  tasks['jvmJar'].configure { t ->
+    // the bnd task conention modifies this jar task accordingly
+    def bndConvention = bndBundleTaskConventionClass.newInstance(t);
+    // modify these lines for MANIFEST.MF properties or for specific bnd instructions
+    bndConvention.bnd = '''
+    Export-Package: okio
+    Automatic-Module-Name: okio
+    Bundle-SymbolicName: com.squareup.okio
+    '''
+    // call the convention when the task has finished to modify the jar to contain OSGi metadata
+    t.doLast {
```

Comment:
One more question: what's different between this approach and typical use of bnd? The way this stuff is hooked up seems pretty atypical! doLast and bndBundleTaskConventionClass both seem awkward.

Reply:
Typically you would apply the plugin itself which will overwrite the jar task and read a file in the projects directory called bnd.bnd where the multiline string should be in. With the build of okio this is not possible because the jar task itself doesn't really do anything the real "jar" task is the "jvmJar" therefore I applied the stuff the plugin does normally to the jar task to the jvmJar task instead.

Reply:
Thanks for the explanation